Blackjack/Controller.cs

```diff
@@ -38,36 +38,44 @@ public Controller(IInput input, List<IPlayer> players, IDeck deck)
         {
             _input = input;
             Players = players;
-            Deck = deck; // may need interface
+            Deck = deck; 
         }
 
         public void Play()
         {
-            var shuffledDeck = Deck.Shuffle();
+            Deck.Shuffle();
             // deal cards - 2 cards to each player
-            Deal(shuffledDeck);
+            DealHand();
+            // this logic is for player
             var input = _input.ReadLine();
             var isValid = Validator.IsValid(input);
-
             while (!isValid)
             {
                 input = _input.ReadLine();
                 isValid = Validator.IsValid(input);
             }
-            ChoiceParser.ParseChoice(input);
+            // need to assign to a variable so player can either hit or stay
+            var choice = ChoiceParser.ParseChoice(input);
+            if (choice == Choice.Hit) DealCard(Players[0], Deck.Cards[0]);
+            // different logic for dealer
         }
 
-        private void Deal(Deck shuffledDeck)
+        private void DealHand()
         {
-            var index = 0;
             foreach (var player in Players)
             {
                 for (int i = 0; i < 2; i++)
                 {
-                    player.ReceiveCard(shuffledDeck.Cards[index]);
-                    index++;
+                    var card = Deck.Cards[i];
+                    DealCard(player, card);
                 }
             }
         }
+
+        private void DealCard(IPlayer player, Card card) // split out so that when player/dealer 'hit' only one card is given
+        {
+            player.ReceiveCard(card);
+            Deck.Cards.Remove(card);

```

Comment:
good. may be worth testing this behaviour i.e. if I start with 52 cards, do I now have 51 cards when a card is dealt.

also, remember law of demeter with the `.Remove()` - it's an implementation detail of `List` so your controller now knows that the Deck Cards prop is a List. Would be nice to have `Deck` implement a public method to remove a card, similar to the `AddCard` method :)